server/src/test/java/org/elasticsearch/cluster/routing/UnassignedInfoTests.java

```diff
@@ -76,10 +82,20 @@ public void testSerialization() throws Exception {
         int failedAllocations = randomIntBetween(1, 100);
```

Comment:
I wonder if we should add a test of bwc serialization? Since the functionality will only be enabled in 7.15 and I am unaware of any shutdown rolling upgrade tests, we could in theory be exposed to a bwc bug in serialization (not that I saw any, looks all good to me).

Reply:
I'm not sure it's worth adding a unit test for this, but I do intend to add some rolling and full-restart upgrade tests for this soon.